### PR TITLE
Clean up llviewermenu

### PR DESCRIPTION
**Pull request by Ansariel**
* Removed a load of void-pointer parameters and replaced actually used ones with actual datatypes
* Removed function definitions without actual implementations
* enable_land_build and enable_object_build were apparently not used anywhere ("VisibleBuild" callback not found in XUI files)
* Removed a bunch of dead, commented out code

### CODE REVIEW

File: indra/newview/llviewermenu.cpp [hunk 1]
```diff
@@ -623,23 +606,23 @@ class LLAdvancedToggleConsole : public view_listener_t
         std::string console_type = userdata.asString();
         if ("texture" == console_type)
         {
-            toggle_visibility( (void*)gTextureView );
+            toggle_visibility(gTextureView);
         }
         else if ("debug" == console_type)
         {
-            toggle_visibility( (void*)static_cast<LLUICtrl*>(gDebugView->mDebugConsolep));
+            toggle_visibility(gDebugView->mDebugConsolep);
```
Comment (nat-goodspeed) [inline indra/newview/llviewermenu.cpp:613]: THANK YOU.

**Review by nat-goodspeed — APPROVED** [review]
I'd like to ask a favor: please try merging with our `release/luau-scripting` branch? We hope to merge that into develop soonish. It doesn't _look_ as though our llviewermenu.cpp changes will conflict, but I can't pull your branch locally to check it myself.

**Review by marchcat — APPROVED** [review]
Thank you, @Ansariel!

**Comment by nat-goodspeed** [comment]
Ansariel, thanks but don't add/change the PR. It makes sense for this one to go into develop before the Lua branch. I was just nervous when I saw the volume of changes in llviewermenu.cpp.

**Comment by Ansariel** [comment]
> Ansariel, thanks but don't add/change the PR. It makes sense for this one to go into develop before the Lua branch. I was just nervous when I saw the volume of changes in llviewermenu.cpp.

The build worked fine without any further manual changes.

**Pull request merged**